commands/exec/index.js

```diff
@@ -127,6 +127,10 @@ class ExecCommand extends Command {
     return runTopologically(this.filteredPackages, runner, {
```

Comment:
It strikes me that it should be the consumers of `runTopologically()` that wrap the `runner` (that they're already creating) with the profiler and output the profiler results, _not_ `@lerna/run-topologically` itself. It's a bit repetitive, but that's nothing new as far as `lerna exec` vs `lerna run` is concerned...

```js
  runCommandInPackagesTopological() {
    let profiler;
    let runner = this.options.stream
      ? pkg => this.runCommandInPackageStreaming(pkg)
      : pkg => this.runCommandInPackageCapturing(pkg);

    if (this.options.profile) {
      profiler = new Profiler(this.options);
      runner = pkg => profiler.run(() => runner(pkg), pkg.name);
    }

    let chain = runTopologically(this.filteredPackages, runner, {
      concurrency: this.concurrency,
      rejectCycles: this.options.rejectCycles,
    });

    if (profiler) {
      chain = chain.then(results => {
        profiler.output(); // changes if it's async, i guess
        return results;
      });
    }

    return chain;
  }
```

Reply:
Also a good point. I'll make the adjustments.